document-store/src/main/java/org/hypertrace/core/documentstore/expression/ArithmeticExpression.java

```diff
@@ -0,0 +1,31 @@
+package org.hypertrace.core.documentstore.expression;
+
+import java.util.List;
+import lombok.Builder;
+import lombok.Singular;
+import lombok.Value;
+
+/**
+ * Expression representing arithmetic operations in a query.
+ *
+ * <p>Example: A-5 can be constructed as <code>
+ *   ArithmeticExpression.builder()
+ *      .operand(LiteralExpression.of("A"))
+ *      .operator(ArithmeticOperator.SUBTRACT)
+ *      .operand(ConstantExpression.of(5))
+ *      .build();
+ * </code> The same can be constructed with different order of operands, as long as the minuend is
+ * specified before the subtrahend E.g.: Another valid ordering could be <code>
+ *   ArithmeticExpression.builder()
+ *     .operator(ArithmeticOperator.SUBTRACT)
+ *     .operand(LiteralExpression.of("A"))
+ *     .operand(ConstantExpression.of(5))
+ *     .build();
+ * </code>
+ */
+@Value
+@Builder
+public class ArithmeticExpression implements Groupable, Projectable, Sortable {
```

Comment:
I'm not sure arithmetic is the right name here. This is really representing any non-aggregate expression as I read it (e.g. `substring(my_col, 0, 3)`)

Reply:
Renamed it as FunctionExpression as "SUBSTRING", "ADD", "SUBTRACT" are all functions. Would be happy to change it if anyone has a better alternative.